core/src/main/java/org/apache/druid/data/input/InputRowSchema.java

```diff
@@ -31,15 +34,38 @@
   private final DimensionsSpec dimensionsSpec;
   private final ColumnsFilter columnsFilter;
 
+  /**
+   * List of metric names for further downstream processing by {@link InputSource}
+   */
+  @Nullable
+  private final List<String> metricNames;
```

Comment:
Should this be a set, or does the order of metricNames matter?

Reply:
It can be a set. The order does not matter